### ISSUE TITLE
Dashboards: Padding not respected when scroll in `y` in the New DS Picker modal 

### ISSUE THREAD

**Issue by juanicabanas** [grafana/grafana#67948]
### 
# What went wrong?

**What happened**:

  -  Because of the fact the div children container has the padding and the scroll behavior, if the content is scrolleable (mostly in mobile), the padding bottom is not respected 
`  <div className={cx(styles.modalContent, contentClassName)}>{children}</div>
`
**What did you expect to happen**:

  - To respect the padding when is scrolled

### 
# How do we reproduce it?

**Step 1**:

- Open a modal with scrolleable content and check it doesn't respect the padding bottom 
![image](https://user-images.githubusercontent.com/11707172/236561597-4d067c1b-1355-4c6b-b0e2-c36a881b52f7.png)
![image](https://user-images.githubusercontent.com/11707172/236561650-ddd92c27-02fa-4f3c-ad64-0ed57e818b57.png)

### 
# What Grafana version are you using?

v9.5.0

### 
### Is the bug inside a Dashboard Panel?

Copy the panel's ["get-help" data](https://grafana.com/docs/grafana/latest/troubleshooting/send-panel-to-grafana-support/) here

### Grafana Platform?

None

### Is this a Regression?

None

**Comment by grafanabot** [comment]
@juanicabanas please add one or more appropriate labels. Here are some tips:

- if you are making an issue, TODO, or reminder for yourself or your team, please add one label that best describes the product or feature area. Please also add the issue to your project board. :rocket:

- if you are making an issue for any other reason (docs typo, you found a bug, etc), please add at least one label that best describes the product or feature that you are discussing (e.g. `area/alerting`, `datasource/loki`, `type/docs`, `type/bug`, etc). [Our issue triage](https://github.com/grafana/grafana/blob/main/ISSUE_TRIAGE.md#3-categorizing-an-issue) doc also provides additional guidance on labeling. :rocket:

 Thank you! :heart:

**Comment by JoaoSilvaGrafana** [comment]
Hey @juanicabanas ! Thanks for raising this, I have checked this in prod and I managed to reproduce it on the Modal that you screenshot but not in other modals. I think this is something to do with the css of the modal content, because when I inspect the current version of the`DataSourceModal`, the right column actually clips the height of its content:
<img width="771" alt="Screenshot 2023-05-09 at 13 51 07" src="https://github.com/grafana/grafana/assets/100691367/8ea096d1-9a7f-479a-8c55-ac4b8c72a3ec">

So the modal is keeping the padding to the height it thinks the content has. If I add `overflow: auto` to the `rightColumn` for example it fixes this, even though I understand this is not ideal.

I think the different flex components might be messing with the heights a bit, so I would suggest playing around with the css to figure out something that works for this case. Let us know how this works for you!
